packages/block-editor/src/components/link-control/index.js

```diff
@@ -339,6 +341,10 @@ function LinkControl( {
 	// Effects
 	const getSearchHandler = useCallback(
 		( val, args ) => {
+			if ( ! showSuggestions ) {
+				return Promise.resolve( [] );
```

Comment:
I've thought a few times about making `URLInput` able to handle non-Promise values by using the `Promise.resolve()` wrapper around the `fetchSuggestions` prop. I wonder what you think to this?

Reply:
I would think this is a good idea as I had a few tries until I realized what's what. This code above is not quire nice, resolving with an empty array, when returning false or the empty array would be much more readable in context.